capstone/capdb/models.py

```diff
@@ -222,36 +235,68 @@ class Court(models.Model):
     name = models.CharField(max_length=255)
     name_abbreviation = models.CharField(max_length=100, blank=True)
     jurisdiction = models.ForeignKey('Jurisdiction', null=True, related_name='%(class)s_jurisdiction', on_delete=models.SET_NULL)
-    slug = models.SlugField(unique=True, max_length=255)
-
-    def __str__(self):
-        return self.slug
-
-
-class Jurisdiction(models.Model):
-    name = models.CharField(max_length=100, blank=True)
-    slug = models.SlugField(unique=True, max_length=255)
-    name_abbreviation = models.CharField(max_length=200, blank=True)
+    slug = models.SlugField(unique=True, max_length=255, blank=False)
 
     def __str__(self):
         return self.slug
 
 
 class CaseXML(models.Model):
-    barcode = models.CharField(max_length=255, unique=True, db_index=True)
+    case_id = models.CharField(max_length=255, unique=True, db_index=True)
     orig_xml = XMLField()
     volume = models.ForeignKey(VolumeXML, related_name='case_xmls')
 
     def __str__(self):
-        return self.barcode
+        return self.case_id
+
+    def update_case_metadata(self):
+        data = get_case_metadata(self.orig_xml)
+        citation, created = Citation.objects.get_or_create(cite=data["citation"],
+                                                           type=["citation_type"])
+
+        case_metadata, created = CaseMetadata.objects.get_or_create(
+            case_id=self.case_id,
+            citation=citation,
+        )
+
+        case_metadata.volume = VolumeMetadata.objects.get(barcode=self.volume.barcode)
+
+        case_metadata.first_page = data["first_page"]
+        case_metadata.last_page = data["last_page"]
+        case_metadata.decision_date_original = data["decision_date_original"]
+        case_metadata.decision_date = data["decision_date"]
+        case_metadata.docket_number = data["docket_number"]
+
+        try:
+            case_metadata.jurisdiction = Jurisdiction.objects.get(name=data["jurisdiction"])
+        except ObjectDoesNotExist:
+            pass
+
+        court_name = data["court"]["name"]
+        court_name_abbreviation = data["court"]["name_abbreviation"]
+
+        if court_name:
+            court, created = Court.objects.get_or_create(name=court_name)
```

Comment:
So the thing is, if two courts have the same name but different jurisdiction, are they the same court? I'm guessing no. So I think jurisdiction wants to be in the `get_or_create()`?

Off the top of my head, I think it's fine to pass in `jurisdiction=case_metadata.jurisdiction` even if `case_metadata.jurisdiction` is `None` -- I don't think you need a special case for that.

BTW a nice little shortcut with this "might exist" stuff is to use `first` instead of `get`:

`case_metadata.jurisdiction = Jurisdiction.objects.filter(name=data["jurisdiction"]).first()`

As long as `name` is unique, that has the same effect of either being the thing if it exists or `None`, but without the try-except.